lib/util/transport.js

```diff
@@ -148,6 +151,42 @@ var qlb_split = function (buffer, print) {
     return ret;
 };
 
+var decode_packed_xml_contents = function(xmlBuf) {
+    var el, partName;
+    var res = {};
```

Comment:
Yep. Instead of returning  `{doc: content, attr: content2}`, use `[{name: "doc", content: content}, {name: "attr", content: content2}]`